npm/index.js

```diff
@@ -6,9 +6,11 @@ var pathFile = path.join(__dirname, 'path.txt')
 function getElectronPath () {
   if (fs.existsSync(pathFile)) {
     var executablePath = fs.readFileSync(pathFile, 'utf-8')
-    if (process.env.ELECTRON_OVERRIDE_DIST_PATH) {
-      return path.join(process.env.ELECTRON_OVERRIDE_DIST_PATH, executablePath)
-    }
+  }
+  if (process.env.ELECTRON_OVERRIDE_DIST_PATH) {
```

Comment:
```js
  if (process.env.ELECTRON_OVERRIDE_DIST_PATH) {
    return path.join(process.env.ELECTRON_OVERRIDE_DIST_PATH, executablePath || 'electron')
  }
  if (fs.existsSync(pathFile)) {
    var executablePath = fs.readFileSync(pathFile, 'utf-8')
    return path.join(__dirname, 'dist', executablePath)
  } else {
      throw ...
```

I think that layout of if statements is more readable

Reply:
Admittedly I haven't given much thought to the structure of this implementation. That said, I think we'd be referencing `executablePath` before it's been initialized with that structure - yeah?

Reply:
Ah yeah, we should just not depend on `path.txt` for resolution.  Use `path.txt` to ensure it's installed but export `getPlatformPath` from `install.js` and use that to get the executable path.

Reply:
@skeggse do you still plan to pursue this PR?

Reply:
It's a feature that no longer impacts me; I would be fine with finishing this, though I guess I don't follow @MarshallOfSound's last comment (maybe I did earlier?)

If I recall my use-case correctly, `path.txt` wouldn't exist because the install script would have been skipped - so checking for it when there's an override doesn't make sense to me. I'm also not clear on what `getPlatformPath` adds - I would expect the `ELECTRON_OVERRIDE_DIST_PATH` variable to be the path containing the binary, but that seems at odds with `getPlatformPath` returning a macos-app-specific string - wouldn't you expect `ELECTRON_OVERRIDE_DIST_PATH=/Applications/Electron.app/Contents/MacOS/Electron` to work?

Reply:
This makes sense to me, I'm good with current implementation.